addons/sale_stock/migrations/11.0.1.0/openupgrade_analysis_work.txt

```diff
@@ -0,0 +1,49 @@
+---Fields in module 'sale_stock'---
+sale_stock   / procurement.group        / sale_id (many2one)            : NEW relation: sale.order
+# Filled in post-migration script
+
+sale_stock   / sale.order               / picking_ids (many2many)       : not a function anymore
+sale_stock   / sale.order               / picking_ids (many2many)       : type is now 'one2many' ('many2many')
+# Nothing to do
+
+sale_stock   / sale.order               / procurement_group_id (many2one): previously in module sale
+# Done, rename references on sale pre-migration script
+
+sale_stock   / sale.order.line          / move_ids (one2many)           : NEW relation: stock.move
+# Nothing to do
+
+sale_stock   / stock.move               / sale_line_id (many2one)       : NEW relation: sale.order.line
+# Filled in post-migration script
+
+sale_stock   / stock.move               / to_refund_so (boolean)        : DEL
+# Nothing to do
+
+sale_stock   / stock.picking            / sale_id (many2one)            : not a function anymore
+# Nothing to do, converted to a related
```

Comment:
It is converted to a related field on the procurement group's sale_id, but this is set in the post script. The field is recomputed during the migration for every picking (which can take some time on a large database) but at that time, the procurement group's sale id is not set in your scripts.
For best performance, precreate both fields in the pre migration script and fill them there using SQL.

Reply:
@StefanRijnhart, you are right. During my migration database to odoo11 I found that there is no previous data to link sale order and stock picking because the procurement group's sale id is not set so I added in the post migration new function to fix this issue : 

`def update_stock_picking_field_from_procurement_group(cr):
    """
    Filling the values of new sale_id field of stock_picking
    with respective values of old procurement_group
    """
    qery = ("UPDATE stock_picking AS sp "
             "SET sale_id = pg.sale_id "
             "FROM procurement_group AS pg "
             "WHERE sp.group_id = pg.id")
    openupgrade.logged_query(
        cr, query, ()
    )
`
@yuriqp  could you please check Stefan review's to approve this module and will be merged. thanks